### PR TITLE
New Checkout: split layout (take 2)

### PR DESCRIPTION
**Pull request by michaeldcain**
Originally attempted in #41063 and reverted in #41363 

This does a few things with the ultimate goal of splitting out a summary into a separate column on desktop screens.

- adds the floated `CheckoutSummary` component to composite-checkout (package, demo, documentation)
- add `CheckoutStepArea` component to handle styling of `CheckoutSteps` + `CheckoutStepBody`
- uses `CheckoutSummary` in WPcom's implementation of composite-checkout
- moves the current site's address to the review step
- adds a default set of purchase features to the summary (per-product features coming in a future PR)
- hides `CheckoutSummary` on mobile screens (collapse mobile toggle coming in future PR)

Before | After
------------ | -------------
before-desktop | after-desktop
before-mobile | after-mobile

Fixes: #41088

**To test:**
- it might help to step through the PR, commit by commit, but no promises that I didn't change my mind later on in the PR
- run the composite-checkout demo with `yarn run composite-checkout-demo`
- verify that minimal CheckoutSummary is displayed
- verify that the demo otherwise works as expected
- from dev or calypso.live, and a plan and/or domain to your cart and visit Checkout with `?flags=composite-checkout-testing`
- verify that on desktop (`>960px`), the checkout summary is displayed, floated to the right, with a list of default features, and the tax (once calculated) and total
- verify that on screens smaller than desktop, the checkout summary is hidden
- verify that Checkout otherwise works as expected

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js [hunk 1]
```diff
@@ -3,46 +3,109 @@
...
+		<>
+			<CheckoutSummaryTitle>{ translate( 'Purchase Details' ) }</CheckoutSummaryTitle>
+			<CheckoutSummaryFeatures>
+				<CheckoutSummaryFeaturesTitle>
+					{ translate( 'Included with your purchase' ) }
+				</CheckoutSummaryFeaturesTitle>
+				<CheckoutSummaryFeaturesList>
+					<CheckoutSummaryFeaturesListItem>
+						<WPCheckoutCheckIcon />
+						{ translate( 'Live chat and email support' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js:29]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Email and live chat support' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/checkout-two-column-layout)

File: client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js [hunk 2]
```diff
@@ -3,46 +3,109 @@
...
+					</CheckoutSummaryFeaturesListItem>
+					<CheckoutSummaryFeaturesListItem>
+						<WPCheckoutCheckIcon />
+						{ translate( 'Free custom domain for a year' ) }
+					</CheckoutSummaryFeaturesListItem>
+					<CheckoutSummaryFeaturesListItem>
+						<WPCheckoutCheckIcon />
+						{ translate( 'Dozens of free themes' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js:37]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Dozens of free themes.' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/checkout-two-column-layout)

File: client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js [hunk 3]
```diff
@@ -3,46 +3,109 @@
...
+					</CheckoutSummaryFeaturesListItem>
+					<CheckoutSummaryFeaturesListItem>
+						<WPCheckoutCheckIcon />
+						{ translate( 'Money back guarantee' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-summary.js:41]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '30-day Money Back Guarantee' )` *ES Score: 9*

**Comment by fditrapani** [comment]
This is looking great! the immediate thing that pops out is that the benefits in the right column are static and not reflecting the products in the cart. This is what I get for a domain only:

image

and this is the business plan:

image

If we want to start with static, that's ok but we'll need to make sure the benefits are generic enough for all our products. 

====

I don't know if this is a regression but I'm upgrading my plan and it's defaulting to the two year length without giving me the choice of switching back to the one year. You can see it in the screenshot above.

**Comment by michaeldcain** [comment]
> If we want to start with static, that's ok but we'll need to make sure the benefits are generic enough for all our products.

@fditrapani: I did this to get it out the door. The most generic are the support and money back points.

> I don't know if this is a regression but I'm upgrading my plan and it's defaulting to the two year length without giving me the choice of switching back to the one year. You can see it in the screenshot above.

We talked about this in Slack, but it's the current Checkout behavior to default to the variation that you currently have selected. I agree that it'd be great to show the other available variations too. I'll look into what'd be required for that.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3349298

Thank you @michaeldcain for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3349298).

**Pull request merged**